python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -280,17 +305,182 @@ def plan_udf_map_op(
     )
 
 
-def _get_udf(op: AbstractUDFMap):
+def plan_download_op(
+    op: Download,
+    physical_children: List[PhysicalOperator],
+    data_context: DataContext,
+) -> MapOperator:
+    assert len(physical_children) == 1
+    input_physical_dag = physical_children[0]
+    compute = get_compute(op._compute)
+    uri_column_name = op.uri_column_name
+
+    fn, init_fn = _get_udf(PartitionActor, (), {}, (uri_column_name,), {})
+    partition_transform_fn = _generate_transform_fn_for_map_batches(fn)
+    partition_transform_fns = [
+        # Convert input blocks to batches.
+        BlocksToBatchesMapTransformFn(
+            batch_size=None,  # TODO: Should we use op._batch_size here?
+            batch_format="pyarrow",
+            zero_copy_batch=True,
+        ),
+        # Apply the PartitionActor function.
+        BatchMapTransformFn(partition_transform_fn),
```

Comment:
Should this just be a `BlockMapTransformFn`? If `partition_transform_fn` returns NumPy batches (which is a valid output batch type), this will break.